src/Sandbox/SourcePolicyInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Sandbox;
+
+use Twig\Source;
+
+/**
+ * Interface for a class that can optionally enable the sandbox mode based on a template's Twig\Source.
+ *
+ * @author Yaakov Saxon
+ */
+interface SourcePolicyInterface
+{
+    public function enableSandbox(Source $source = null): bool;
```

Comment:
why making the argument optional ?

Reply:
Good point. My intention was to mark it as nullable, since the upstream method Source comes from is marked as = null. I didn't realize PHP has a distinct syntax for marking as required but nullable. I'll fix that.